src/Card.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { memo, forwardRef } from "react";
+import type { ReactNode } from "react";
+import { symToStr } from "tsafe/symToStr";
+import { assert } from "tsafe/assert";
+import type { Equals } from "tsafe";
+
+import { FrIconClassName, RiIconClassName } from "./lib/generatedFromCss/classNames";
+import { fr, RegisteredLinkProps } from "./lib";
+import { getLink } from "./lib/routing";
+import { cx } from "./lib/tools/cx";
+
+import "./dsfr/component/card/card.css";
+
+//https://main--ds-gouv.netlify.app/example/component/card/
+export type CardProps = {
+    className?: string;
+    title: ReactNode;
+    linkProps: RegisteredLinkProps;
+    desc?: ReactNode;
+    imageUrl?: string;
+    imageAlt?: string;
+    start?: ReactNode;
+    detail?: ReactNode;
+    end?: ReactNode;
+    endDetail?: ReactNode;
+    badges?: ReactNode[]; // todo: restrict to badge component ? these badges are display on the image
+    footer?: ReactNode; // where actions can be placed
+    horizontal?: boolean;
+    size?: "small" | "medium" | "large"; // only affect the text
+    enlargeLink?: boolean; // make the whole card clickable
+    iconId?: FrIconClassName | RiIconClassName; // only needed when enlargeMode=true
+    shadow?: boolean;
+    background?: boolean;
+    border?: boolean;
+    grey?: boolean;
+    classes?: Partial<
+        Record<
+            | "root"
+            | "title"
+            | "card"
+            | "link"
+            | "body"
+            | "content"
+            | "desc"
+            | "header"
+            | "img"
+            | "imgTag"
+            | "start"
+            | "detail"
+            | "end"
+            | "endDetail"
+            | "badges"
+            | "footer",
+            string
+        >
+    >;
+} & (CardProps.Default | CardProps.Horizontal);
+
+export namespace CardProps {
+    export type Default = {};
+
+    export type Horizontal = {
+        horizontal: true;
+    };
+}
+
+/** @see <https://react-dsfr-components.etalab.studio/?path=/docs/components-alert> */
+export const Card = memo(
+    forwardRef<HTMLDivElement, CardProps>((props, ref) => {
+        const {
+            className,
+            title,
+            linkProps,
+            desc,
+            imageUrl,
+            imageAlt,
+            start,
+            detail,
+            end,
+            endDetail,
+            badges,
+            footer,
+            horizontal = false,
+            size = "medium",
+            classes = {},
+            enlargeLink = true,
+            background = true,
+            border = true,
+            shadow = false,
+            grey = false,
+            iconId,
+            ...rest
+        } = props;
+
+        assert<Equals<keyof typeof rest, never>>();
+
+        const { Link } = getLink();
+
+        return (
+            <div
+                className={cx(
+                    fr.cx("fr-card"),
+                    enlargeLink === true && fr.cx("fr-enlarge-link"),
```

Comment:
When it's a boolean we don't need to `x === true`, `bool` will do it.  
It's when x isn't a boolean that I prefer explicitely checking for the `undefined`